### DISCUSSION TITLE
Building my own custom device integration with yaml files- without a support/pull request

### DISCUSSION THREAD

**Discussion by dustball62** [make-all/tuya-local#1235] [Q&A] [answered]
Hello- I truly hate to bother you as I see you are swamped with requests and discussions. I don't want to add to your work loc So I poured over you Python Code and Yaml files to find out when a place a new YAML file in the custom_components\Tuya_local\Devices folder its not picked up by the integration.
I'm trying to add an alarm system and I have the DP codes and product/ID, etc, and I used your existing alarm file -gautone_pg103_alarm.yaml-and built in what should be in the new file. I didn't run test_device_config.py because I really didn't need to change that much and left in what was only valuable.

Your Python code seems like it scans the directory and determines if there's a match, I don't see what else I need to do place to get the integration to pick it up. I'm still getting the:

"Sorry, there is no support for this device."

Message.

Is the file name itself that important? 
There was another thread that was similar to this one but didn't get to core of my issue.
I'm hoping your answer is simple and this will help others takes some work load off of you and your "team",

**Comment by make-all** [comment]
Only the yaml extension is important.  More likely you have made your config from iot.tuya.com docs, but it is not exactly matching the actual data that the device is sending.  There should be a log message showing a list of dps along with that "Sorry there is no support for this device" popup, and you can compare the types with the types in your config, and look for any that are missing from the log, which need to be marked `optional: true` in the config.

**Reply by dustball62** [reply]
Ok thanks I'll look at that. I know I didn't include all the DPs because some I was exactly sure the need/purpose. One in particular off the top of my head was for "adding accessories" - which believe is a for setting up new sensors(motion detectors and the like).  I any case I'll go back and see if I can locate the log and see what that tells me.

**Reply by dustball62** [reply]
ok I went through and added the missing DPs from the log and that fixed it. It was added selected the right one. Now I can add the smoke detector that is also missing. Greatly appreciate your time and help

**Comment by dustball62** [comment, marked as answer]
Fixed by reviewing the log and adding the missing DPs in the Yaml file for the new entity.
